KPI-team1-frontend/src/pages/KpiModalPage/KpiValuesModaleSection.tsx

```diff
@@ -0,0 +1,285 @@
+            rows={3}
+            placeholder="Optional: add comment to your changes"
+            onChange={(e) => {
+              setComment(e.target.value);
+            }}
+          />
+        </label>
+        <div className="pt-4 flex justify-end">
+          <button className="w-28 h-10 mr-4 bg-white rounded border border-customYellow justify-center items-center gap-2 inline-flex text-zinc-700 text-base font-medium">
+            Cancel
+          </button>
+          <button
+            className="w-28 h-10 bg-customYellow rounded justify-center items-center gap-2 inline-flex text-base font-medium"
+            onClick={() => handleSave(newValue, newDate, comment)}
+          >
+            Save
+          </button>
+        </div>
+      </>
+    );
+  };
+  const renderPreviousValues = () => {
+    return (
+      <>
+        <div className="mt-4 text-2xl">Previous Values</div>
+        <div className="">
+          <DataGrid
+            slots={{
+              loadingOverlay: LinearProgress,
+            }}
+            loading={isLoading}
+            getRowId={(row) => row.kpi_value_history_id}
+            rows={kpiValues}
+            rowSelection={false}
+            columns={PREVIOUS_VALUES_COLUMNS}
+            classes={{
+              columnHeaders: "bg-customPurple",
+              columnHeader: "uppercase",
+            }}
+            {...other}
+          />
+        </div>
+      </>
+    );
+  };
+  const renderSetTargetValue = () => {
+    const handleSaveTarget = async (value: string | null) => {
+      if (!value) return;
+      const numberedValue = Number(value);
+      if (numberedValue === kpi.target_value) return;
+      try {
+        const { data, error } = await supabase
+          .from("target")
+          .upsert({
+            target_id: kpi.target_id ? kpi.target_id : undefined,
+            kpi_id: kpi.kpi_id,
+            circle_id: circleId,
+            target_value: numberedValue,
+          })
+          .select("target_value");
+        if (data) {
+          const result = data[0].target_value;
+          alert(`Target value successfully updated to ${result}`);
+          setTargetValue(result);
+        }
+        if (error) {
+          alert(`Error updating the target \nDetails: ${error.message}`);
+          return;
+        }
+        fetchKpiDefinition();
+      } catch (error: any) {
+        console.log(error.message);
+      }
+    };
+
+    return (
+      <>
+        <div className="mt-4 text-2xl">Set a target value for this year</div>
+        <div className="flex justify-between my-2">
+          <label className="font-medium w-full">
+            Enter a target value
+            <input
+              className="block w-full font-normal text-neutral-800 p-2 border rounded-md border-neutral-400"
+              name="target value"
+              type="number"
+              placeholder="What's your target"
+              onBlur={(event: React.FocusEvent<HTMLInputElement>) => {
```

Comment:
Hmmm, not sure about that one in particular about the alert on successful save, was this discussed with Simona?

Reply:
I have just ask her and she was ok with it 😅